src/Sarif.Driver/Sdk/PluginDriverCommand.cs

```diff
@@ -37,6 +42,102 @@
             return assemblies;
         }
 
+        public bool ValidateExportUri(string outputFilePath, string exportUri)
+        {
+            if (string.IsNullOrEmpty(exportUri))
+            {
+                return true;
+            }
+
+            // If exportUri is not empty, then outputFilePath must exist.
+            if (string.IsNullOrEmpty(outputFilePath))
+            {
+                return false;
+            }
+
+            return Uri.IsWellFormedUriString(exportUri, UriKind.Absolute);
+        }
+
+        public bool ValidateFile(IAnalysisContext context, string filePath, string policyName, bool? shouldExist)
+        {
+            if (filePath == null || filePath == policyName) { return true; }
+
+            Exception exception = null;
+
+            try
+            {
+                bool fileExists = FileSystem.FileExists(filePath);
+
+                if (fileExists || shouldExist == null || !shouldExist.Value)
+                {
+                    return true;
+                }
+
+                Errors.LogMissingFile(context, filePath);
+            }
+            catch (IOException ex) { exception = ex; }
+            catch (SecurityException ex) { exception = ex; }
+            catch (UnauthorizedAccessException ex) { exception = ex; }
+
+            if (exception != null)
+            {
+                Errors.LogExceptionAccessingFile(context, filePath, exception);
+            }
+
+            return false;
+        }
+
+        public bool ValidateFiles(IAnalysisContext context, IEnumerable<string> filePaths, string policyName, bool shouldExist)
```

Comment:
>bool [](http://example.com/codeflow?start=15&length=4)

I'm seeing lots of public helpers in this class. Many of them look like they should be static methods. I am not seeing any granular unit tests for these classes.

First, please make sure you are adding minimal code to our SDK. Second, make sure that your accessibility is correct, either public or internal. Make sure things are static that should be. Write unit tests and report your coverage to me.

Reply:
Those methods were in the AnalyzeCommandBase/MultithreadCommandBase (both had the same methods). When I moved I didn't change their accessibility.

Now, I'm changing to internal or internal static.

thanks for the suggestions :)